state/conn_test.go

```diff
@@ -128,3 +128,14 @@ func (s *ConnSuite) NewStateForModelNamed(c *gc.C, modelName string) *state.Stat
 	s.AddCleanup(func(*gc.C) { otherState.Close() })
 	return otherState
 }
+
+// SetJujuManagementSpace mimics a controller having been configured with a
+// management space. It is the space name that constrains the set of
+// addresses agents should use for controller communication.
+func (s *ConnSuite) SetJujuManagementSpace(c *gc.C, space string) {
+	controllerSettings, err := s.State.ReadSettings("controllers", "controllerSettings")
```

Comment:
I believe we will ultimately want to have "juju controller-config" be able to set values in controller config, the way we can currently do with "juju model-config".
Would that provide you with a better API here than ReadSettings()... Write() ?